src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,45 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $border-width;
+  padding: $padding;
+  width: $column-width;
+  margin: 0 auto;
+}
+
+.calendar__day {
+  width: $day-size;
+  height: $day-size;
+  border: $border-width solid black;
+  background-color: #eee;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  position: relative;
+  transition: background-color, transform, 0.5s;
+
+  &:hover {
+    cursor: pointer;
+    background-color: pink;
+    transform: translateY(-20px);
+  }
+
+  @for $days from 1 through 31 {
+    &:nth-child(#{$days})::before {
+      content: '#{$days}';
+      position: absolute;
+    }
+  }
+}
+
+@each $day, $day-position in $week-days {
```

Comment:
The variable `$week-days` is used here, but it's not defined within this file. Ensure that it is defined in an imported SCSS file or within this file.